dra-evolution/pkg/api/claim_types.go

```diff
@@ -1,379 +1,464 @@
 package api
 
 import (
-	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 // DeviceClass is a vendor or admin-provided resource that contains
-// contraint and configuration information. Essentially, it is a re-usable
-// collection of predefined data that device claims may use.
+// device configuration and requirements. It can be referenced in
+// the device requests of a claim to apply these presets.
 // Cluster scoped.
 type DeviceClass struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
-	Spec DeviceClassSpec `json:"spec,omitempty"`
+	// Config defines configuration parameters that apply to each request in a claim using this class.
```

Comment:
Can we clarify this?  I think I misunderstood the meaning, wrote a comment, and then realized I had to be wrong.

I think you meant something like:

"""
Config defines configuration parameters that apply to each device which is claimed via this class
"""

IOW, if my claim has 2 requests, with 2 classes, this only applies to those devices which use THIS class, not all devices.  I think this is what you meant, but it isn't as clear as it could be :)